packages/compass-aggregations/src/modules/pipeline-builder/pipeline-builder.ts

```diff
@@ -118,7 +119,11 @@ export class PipelineBuilder {
     namespace: string,
     options: PreviewOptions
   ): Promise<Document[]> {
-    const pipeline = this.getPipelineFromSource();
+    // For preview we ignore $out/$merge stage.
+    const pipeline = [...this.getPipelineFromSource()];
+    if (isLastStageOutputStage(pipeline)) {
+      pipeline.splice(pipeline.length - 1, 1);
```

Comment:
This can be just a pop
```suggestion
      pipeline.pop();
```